### PR TITLE
Added support for empty list of nodal attributes for fort.15

### LINKED ISSUE #1272: Enable `control_file_gen.pl` to handle an empty list of nodal attributes to activate in `fort.15`

### LINKED ISSUE #1273: Remove redundant platform specifications for debian and desktop
The desktop machines `kitt` and `soldier` resolve as the `desktop` platform, which overlaps with `debian`. These redundant platform specifications should be removed.

### PR DESCRIPTION
**Pull request by jasonfleming**
resolves #1272 resolves #1273

### CODE REVIEW

File: control_file_gen.pl [hunk 1]
```diff
@@ -610,17 +617,20 @@
...
+   my $numLines = 0;
    while(<$nafi>) {
       foreach my $key (keys %{$p->{nodal_attributes}->{default_values}}) {
          my $tag = "%"."$key"."_default"."%";
          my $value = "$p->{nodal_attributes}->{default_values}->{$key}";
          s/$tag/$value/;
       }
       print $nafo $_;
+      $numLines++;
    }
```
Comment (wwlwpd) [inline control_file_gen.pl:629]:
```suggestion
   # accumulate first 50 block of lines
   my $numLines = 0;
   my $first50lines;
   for my $line (1 .. 50) {
     $first50lines .= <$nafi>;
   }

   # global s///
   foreach my $key (keys %{$p->{nodal_attributes}->{default_values}}) {
     my $tag   = "%"."$key"."_default"."%";
     my $value = $p->{nodal_attributes}->{default_values}->{$key};
     $first50lines =~ s/$tag/$value/g;
   }

   print $nafo $first50lines;

   foreach my $line (<$nafi>) {
     print $nafo $line;
   }
```

Comment (jasonfleming) [reply control_file_gen.pl]: Thank you @wwlwpd I have now implemented this and tested it successfully.

**Comment by jasonfleming** [comment]
Also resolves #1276

**Comment by jasonfleming** [comment]
This is ready for review and merge (if approved).

**Review by wwlwpd — APPROVED** [review]
Looks good to me. Thanks for removing stale/duplicate code from platforms.sh

**Pull request merged**